### DISCUSSION TITLE
Show meter readings history in high granularity (15-minutes intervals)

### DISCUSSION THREAD

**Discussion by maorcc** [GuyKh/iec-custom-component#125] [Ideas]
The IEC's RemoteReadingRange API enables detailed electric consumption readings at 15-minute intervals.

The current integration in Home Assistant only shows the meter reading in a "daily" granularity in the `latest_meter_reading` entity. 

Can we add an entity with high granularity consumption with no updates shown for the current day?

I know that the data is only received once a day. Does Home Assistant not support retroactive sensor data updates from the past 24 hours? If this is a Home Assistant limitation, perhaps we should consider submitting a feature request to HA. It's a reasonable expectation for a sensor that communicates through the cellular network.

**Comment by GuyKh** [comment]
Good question, the problem, or complexity comes from two facts:
1. Like you mentioned, HA doesn't allow adding *historic* sensor data, i.e. You can only set thr current value for a sensor and update it as new data flows in. 
2. Unlike Sensors, there's this concept of LongTerm Statistics (https://data.home-assistant.io/docs/statistics/ and https://developers.home-assistant.io/docs/core/entity/sensor/#long-term-statistics), which can be added from sensor data or manually (this happens on this integration). The main advantages of this is that your can add historic data and the fact that the data is ammune to database prune (i.e. Stays logged). However, the down side is that the data is aggregated in a hourly level (no less) and so, despite having a granularity of 15 min data, in this component I do aggregation of a full hour.


I hope this clarifies the reasons and limitations.

**Reply by maorcc** [reply]
On which entity can I see the long-term statistics with hourly resolution data?  

Are the long-term statistics updated daily?

**Reply by GuyKh** [reply]
You can see it in the Energy dashboard.

Note that some sensors can report LTS (in addition to latest data) but AFAIK there's no component to present it.

**Reply by maorcc** [reply]
I don't see hourly IEC data on the Energy dashboard.  Just daily.
![image](https://github.com/GuyKh/iec-custom-component/assets/255973/84cbf43d-4068-4747-84a5-2bc1b4c1b1d6)

**Reply by GuyKh** [reply]
It's about which entity you select.

You show here the Sensor.*, select the iec:* one

**Reply by maorcc** [reply]
Which IEC one? I have only 3 to choose from, and I don't see LTS on any of them:

![image](https://github.com/GuyKh/iec-custom-component/assets/255973/cbf64476-c549-4265-9a04-a9e247a9e5a6)

**Reply by GuyKh** [reply]
![image](https://github.com/GuyKh/iec-custom-component/assets/3136012/e18d2bd4-b7f6-4f6b-ae8b-9702c7e8457a)
This one

**Reply by maorcc** [reply]
Strange. I don't have such an entity.  Can we try debugging it together? I think I sent you a connection request on Discord.

**Reply by maorcc** [reply]
It works now.

The issue is that this IEC integration does not work right during the first month following the smart meter installation. It is important to mention this limitation in the README.md because it is likely that people will try this integration right after having their smart meter installed.

**Comment by maorcc** [comment]
One more place that I think shows the LTS information is the [history-explorer-card](https://github.com/SpangleLabs/history-explorer-card). But I cannot find the IEC LTS information with it right now. @GuyKh, is it because the TLS is not part of another entity?

**Reply by GuyKh** [reply]
@maorcc I want to emphasize a difference between two:
1. An entity with a long-time statistics - You can enable LTS for an entity which causes data aggregation.
2. Long Time Statistics without an entity related to it - You can also manually send LTS data without a related entity. This is what the integration does (as the only entity I think is relevant is the `last-meter-reading` where the current data vs the aggregated one would cause errors).

From a quick glimpse at the card, it uses entities rather than LTS data directly, but I might be wrong.
I also wasn't able to find it in HACS
